tests/Fixer/PhpUnit/PhpUnitTestAnnotationFixerTest.php

```diff
@@ -0,0 +1,427 @@
+{
+    /**
+     */
+    public function testWorks() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works() {}
+}',
+            ],
+            'Annotation is removed, the function is one word and we want it to use snake case' => [
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     */
+    public function test_works() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works() {}
+}',
+                ['case' => 'snake'],
+            ],
+            'Annotation is added, and it is snake case' => [
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function it_has_snake_case() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    public function test_it_has_snake_case() {}
+}',
+                ['style' => 'annotation'],
+            ],
+            'Annotation is removed, and it is snake case' => [
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     */
+    public function test_it_has_snake_case() {}
+
+    public function helper_function() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function it_has_snake_case() {}
+
+    public function helper_function() {}
+}',
+                ['case' => 'snake'],
+                ],
+                'Annotation gets added, it has an @depends, and we use snake case' => [
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works_fine () {}
+
+    /**
+     * @test
+     * @depends works_fine
+     */
+    public function works_fine_too() {}
+}',
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    public function test_works_fine () {}
+
+    /**
+     * @depends test_works_fine
+     */
+    public function test_works_fine_too() {}
+}',
+                    ['style' => 'annotation'],
+                    ],
+                'Annotation gets removed, it has an @depends and we use camel case' => [
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     */
+    public function test_works_fine () {}
+
+    /**
+     * @depends test_works_fine
+     */
+    public function test_works_fine_too() {}
+}',
+                        '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works_fine () {}
+
+    /**
+     * @depends works_fine
+     */
+    public function works_fine_too() {}
+}',
+                    ['case' => 'snake'],
+                ],
+            'Class has both camel and snake case, annotated functions and not, and wants to add annotations' => [
```

Comment:
this is pretty nice test case I was thinking about, big :+1: you added it already :)

yet, let us extend a bit, as I found edge case that is not properly handled
https://github.com/BackEndTea/PHP-CS-Fixer/pull/1